lib/utils/visUtil.cpp

```diff
@@ -341,19 +347,42 @@ double StatTracker::get_min() {
 }
 
 double StatTracker::get_avg() {
+    std::lock_guard<std::mutex> lock(tracker_lock);
+
     if (count == 0) {
         return NAN;
     }
     return avg;
 }
 
 double StatTracker::get_std_dev() {
+    std::lock_guard<std::mutex> lock(tracker_lock);
+
     if (count <= 1) {
         return NAN;
     }
     return std_dev;
 }
 
+nlohmann::json StatTracker::get_json() {
+    std::lock_guard<std::mutex> lock(tracker_lock);
+
+    nlohmann::json tracker_json = {};
+    tracker_json["unit"] = unit;
+    tracker_json["is_optimized"] = is_optimized;
+    tracker_json["size"] = buf_size;
```

Comment:
I'm also not sure we need this either.   The length of the array below gives most of this information implicitly.  Knowing the max length of the tracker doesn't add much for the viewer.